clients/src/test/java/org/apache/kafka/test/TestUtils.java

```diff
@@ -267,11 +267,13 @@ public static void waitForCondition(final TestCondition testCondition, final Str
     public static void waitForCondition(final TestCondition testCondition, final long maxWaitMs, String conditionDetails) throws InterruptedException {
         final long startTime = System.currentTimeMillis();
 
-        while (!testCondition.conditionMet() && ((System.currentTimeMillis() - startTime) < maxWaitMs)) {
+        boolean testConditionMet = testCondition.conditionMet();
+        while (!testConditionMet && ((System.currentTimeMillis() - startTime) < maxWaitMs)) {
```

Comment:
Also, it may be worth adding a comment explaining why it's important to avoid evaluating the condition more than once (this will help ensure the behaviour is not changed by someone else).